NSubstitute.DbConnection/NSubstitute.DbConnection/MockQuery.cs

```diff
@@ -95,7 +143,7 @@ public bool Matches(IDbCommand command)
             {
                 if (!(command.Parameters[i] is DbParameter parameter) ||
                     !Parameters.TryGetValue(parameter.ParameterName, out var parameterValue) ||
```

Comment:
Existing naming here is pretty sketchy, to the extent that it's not super clear where `parameter` refers to the mock query or the command that are being matched with each other. This gets worse now that the `out var parameterValue` is actually the entire mock parameter and not just the parameter value.

Maybe rename `parameter` to be `commandParameter` and `parameterValue` to be `mockParameter`?